src/camera.rs

```diff
@@ -19,34 +19,51 @@
 use ggez;
 use ggez::GameResult;
 use ggez::graphics;
-use na;
-use Vector2;
+use {Point2, Vector2, Matrix3, Similarity2, Translation2, Projective2};
+use na::UnitComplex;
 
-// Hmm.  Could, instead, use a 2d transformation
-// matrix, or create one of such.
+// Now uses Similarity and Projective transforms
 pub struct Camera {
-    screen_size: Vector2,
-    view_size: Vector2,
-    view_center: Vector2,
+    transform: Similarity2,
+    screen_transform: Projective2
 }
 
 impl Camera {
     pub fn new(screen_width: u32, screen_height: u32, view_width: f64, view_height: f64) -> Self {
         let screen_size = Vector2::new(screen_width as f64, screen_height as f64);
         let view_size = Vector2::new(view_width as f64, view_height as f64);
+        let units_per_pixel = view_size.component_div(&screen_size);
+        // Similarities only support uniform scaling
+        assert_eq!(units_per_pixel.x, units_per_pixel.y);
```

Comment:
This should be error handled in some way, possibly add a `fit` parameter that would determine whether to use the smaller or larger for fit/fill frame behavior.